packages/nodes-base/credentials/GristApi.credentials.ts

```diff
@@ -0,0 +1,49 @@
+import {
+	ICredentialType,
+	INodeProperties,
+} from 'n8n-workflow';
+
+export class GristApi implements ICredentialType {
+	name = 'gristApi';
+	displayName = 'Grist API';
+	documentationUrl = 'grist';
+	properties: INodeProperties[] = [
+		{
+			displayName: 'API Key',
+			name: 'apiKey',
+			type: 'string',
+			default: '',
+		},
+		{
+			displayName: 'Plan Type',
+			name: 'planType',
+			type: 'options',
+			default: 'free',
+			options: [
+				{
+					name: 'Free',
+					value: 'free',
+				},
+				{
+					name: 'Paid',
+					value: 'paid',
+				},
+			],
+		},
+		{
+			displayName: 'Custom Subdomain',
+			name: 'customSubdomain',
+			type: 'string',
+			default: '',
+			required: true,
```

Comment:
Not sure if this is a bug in n8n, this says it's required but it's not actually forcing me to fill it in, and of course leaving it blank leads to other errors.